scripts/generate-doc.sh

```diff
@@ -0,0 +1,107 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Shell script to generate the Terraform documentation for the resource and data sources.
+#
+# Usage: ./generate-doc.sh" ${resource_name}
+#   resource_name is the terraform resource name. Example: search_deployment
+#   echo "Examples:"
+#   echo "  search_deployment"
+#   echo "  project"
+#   echo "  online_archive"
+#   echo "  encryption_at_rest"
+#
+# The scripts requires to install tfplugindocs and to create the resource templates in 
+# templates/resources/${resource_name}.html.markdown.tmpl and 
+# templates/data-sources/${resource_name}.html.markdown.tmpl
+# templates/data-sources/${resource_name}s.html.markdown.tmpl
+
+set -euo pipefail
+
+TF_VERSION="${TF_VERSION:-"1.6.6"}" # TF version to use when running tfplugindocs. Default: 1.6.6
+TEMPLATE_FOLDER_PATH="${TEMPLATE_FOLDER_PATH:-"templates"}" # PATH to the templates folder. Default: templates
+
+
+# if [ -z "${resource_name}" ]; then
+if [ $# -eq 0 ]; then
+    echo "Error: Input param not found"
+    echo "Usage: ./generate-doc.sh resource_name"
+    echo "resource_name is the terraform resource and data source name."
+    echo "Examples:"
+    echo "  search_deployment"
+    echo "  project"
+    echo "  online_archive"
+    echo "  encryption_at_rest"
+    exit 1
+fi
+
+resource_name="$1"
+
+if [ ! -f "${TEMPLATE_FOLDER_PATH}/resources/${resource_name}.html.markdown.tmpl" ]; then
+    echo "Error: we coudn't find the template for the ${resource_name} resource"
+    echo "Please, make sure to include the resource template under ${TEMPLATE_FOLDER_PATH}/resources/${resource_name}.html.markdown.tmpl"
+    exit 1
+fi
+
+if [ ! -f "${TEMPLATE_FOLDER_PATH}/data-sources/${resource_name}.html.markdown.tmpl" ]; then
+    echo "Error: we coudn't find the template for the ${resource_name} data source"
+    echo "Please, make sure to include the data source template under ${TEMPLATE_FOLDER_PATH}/data-sources/${resource_name}.html.markdown.tmpl"
+    exit 1
+fi
+
+if [ ! -f "${TEMPLATE_FOLDER_PATH}/data-sources/${resource_name}s.html.markdown.tmpl" ]; then
+    echo "Warning: we coudn't find the template for the ${resource_name}s data source"
+    echo "Please, make sure to include the data source template under ${TEMPLATE_FOLDER_PATH}/data-sources/${resource_name}.html.markdown.tmpl"
+    printf "Skipping this check: We assume that the resource does not have a plural data source.\n\n"
+fi
+
+# tfplugindocs uses this folder to generate the documentations
+mkdir -p docs
+
+tfplugindocs generate --tf-version "${TF_VERSION}" --website-source-dir "${TEMPLATE_FOLDER_PATH}"
+
+if [ ! -f "docs/resources/${resource_name}.html.markdown" ]; then
```

Comment:
I prefer leaving the code as it is as the last `if` has a different behaviour and does not return an error if the template is missing since some resources do not have the plural form of the data source. Happy to create a follow-up PR if you have a strong opinion about this change. Thanks for the feedback!